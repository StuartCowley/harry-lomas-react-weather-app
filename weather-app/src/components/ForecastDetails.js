import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__min-temperature">
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__max-temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__humidity">{humidity}%</div>
      <div className="forecast-details__wind">
        {wind.speed}mph, {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};