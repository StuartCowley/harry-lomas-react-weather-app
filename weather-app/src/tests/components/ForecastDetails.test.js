import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 54,
    wind: {
      speed: 5,
      direction: "w",
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("12°C")).toHaveClass("forecast-details__min-temperature");
    expect(getByText("22°C")).toHaveClass("forecast-details__max-temperature");
    expect(getByText("54%")).toHaveClass("forecast-details__humidity");
    expect(getByText("5mph, w")).toHaveClass("forecast-details__wind");
  });
});
