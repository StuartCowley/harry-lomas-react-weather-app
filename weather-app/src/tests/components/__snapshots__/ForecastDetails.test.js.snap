// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      12°C
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      54%
    </div>
    <div
      class="forecast-details__wind"
    >
      5mph, w
    </div>
  </div>
</DocumentFragment>
`;
